### PR TITLE
Parse date fix h2and presto

### PR DESCRIPTION
**Pull request by vj2050gs**
#### What type of PR is this?

Fix 

#### What does this PR do / why is it needed ?

parseDate is a pure function that is mapped to parsedatetime and date_parse in H2 and Presto databases.
Fixed this pure function to correctly generate SQL for these DBs.

### CODE REVIEW

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure [hunk 1]
```diff
@@ -81,7 +81,7 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::h2::g
     dynaFnToSql('monthNumber',            $allStates,            ^ToSql(format='month(%s)')),
     dynaFnToSql('mostRecentDayOfWeek',    $allStates,            ^ToSql(format='dateadd(DAY, case when %s - DAY_OF_WEEK(%s) > 0 then %s - DAY_OF_WEEK(%s) - 7 else %s - DAY_OF_WEEK(%s) end, %s)', transform={p:String[1..2] | $p->formatMostRecentH2('current_date()')}, parametersWithinWhenClause = [false, false])),
     dynaFnToSql('now',                    $allStates,            ^ToSql(format='current_timestamp()')),
-    dynaFnToSql('parseDate',              $allStates,            ^ToSql(format='parsedatetime(%s,%s)')),
+    dynaFnToSql('parseDate',              $allStates,            ^ToSql(format='parsedatetime(%s,%s)', transform={p:String[*]|$p->transformParseDateParams()})),
```
Comment (epsstan) [inline legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure:84]: Should this be String[2] ?

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure [hunk 2]
```diff
@@ -157,6 +157,14 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::h2::c
       'cast( parseDateTime('+$params->at(0)+','+$dateFormat +') as date)';
 }
 
+function <<access.private>> meta::relational::functions::sqlQueryToString::h2::transformParseDateParams(params:String[*]):String[2]
```
Comment (epsstan) [inline legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure:160]: Think these tests should be organized according to new scheme by @vikask-gs.

@vikask-gs - Have we written down (in Legend), how new tests should be organized ?

**Review by epsstan — CHANGES_REQUESTED** [review]

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure (hunk 2 above)
Comment (vikask-gs) [reply legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure:160]: yes, communicating with Vaishnavi offline on this

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure (hunk 1 above)
Comment (vj2050gs) [reply legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure:84]: ideally it should be String[1..2] as we will have a date parameter as string, and a date format which is optional

**Pull request closed without merging**